### ISSUE TITLE
Feature Request: make fonts display correct symbols instead of squares

### ISSUE THREAD

**Issue by Tinase-nau** [notepad-plus-plus/notepad-plus-plus#5465]
### Description of the Issue
Changed font to Nirmala UI. Certain box drawing Unicode characters (U+2553, U+2556, U+2551, U+2559, U+255C, and perhaps other Unicode characters) appear as squares instead. These characters display fine in Microsoft Windows Notepad though. (Perhaps even other fonts are affected.)
So perhaps Nirmala UI doesn't have support for such characters? But how does Microsoft Windows Notepad achieve it when I use the same font there? That's why I don't know if it's a bug, or additional feature implementation is required...

### Steps to Reproduce the Issue

1. Change Notepad++ font to Nirmala UI
2. Copy-paste affected characters into N++ from
https://en.wikipedia.org/wiki/Box-drawing_character#Unicode

### Expected Behavior
Expected to see correct characters with Nirmala UI font.

### Actual Behavior
Got squares instead.

### Debug Information
Notepad++ v7.6.4   (32-bit)
Build time : Mar  6 2019 - 02:52:22
Path : J:\Program Files (x86)\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS : Windows 7 (64-bit)
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExport.dll

**Comment by sasumner** [comment]
>But how does Microsoft Windows Notepad achieve it when I use the same font there?

Read about the font-fallback mechanism notepad.exe uses, here: https://notepad-plus-plus.org/community/topic/16497/missing-unicode-characters-in-notepad

**Comment by Tinase-nau** [comment]
I presume N++ won't be getting font-fallback mechanism?

**Comment by sasumner** [comment]
>I presume N++ won't be getting font-fallback mechanism?

Well, maybe change this issue slightly to be a definitive request for it, and leave it open, and...we will see! :)

**Comment by Tinase-nau** [comment]
> > I presume N++ won't be getting font-fallback mechanism?
> 
> Well, maybe change this issue slightly to be a definitive request for it, and leave it open, and...we will see! :)

There, I fixed it.

**Comment by Bison700** [comment]
I also would like to have a font fallback mechanism. Most other apps support it.
Windows has .compositefont files that contain fallback fonts that are accesible via an API: https://fileinfo.com/extension/compositefont

**Comment by mergen3107** [comment]
May I join to this thread? I agree, a fallback mechanism would be great for Notepad++ ! Especially, when creating new keyboard layouts for KOReader and Android keyboards.

**Comment by fahrrad13** [comment]
I agree as well, that a fallback font option would be very neat for various purposes.

**Comment by mnh48** [comment]
I want to request this as well, if notepad++ would have option to just load another font when current font does not have the characters, it would save me headache from needing to write them somewhere else and copy and paste them into notepad++

<img width="566" height="417" alt="Image" src="https://github.com/user-attachments/assets/69792b5b-288c-4762-93ed-6dc2cbac0f04" />

majority of fonts out there doesn't have support for niche community and scripts from aboriginal languages, but if I choose the font that support these niche scripts then other language (such as CJKV) stop working as the fonts used by the niche community don't support them, but I write stuff that requires multiple language to work at the same time

it would be nicer if we can specify specific font for notepad++ to fallback to so that they would appear correctly

---
edit: here's sample text for people to play around with fonts
```
ꤰꥈꤳꥎ ꤳꥈꥐ ꤾꥁꥉꥑ ꤸꥎꥑꤴꥉꤰ, ꤳ꥓ꤸꥈꥆꥐ ꥁꥋꤰ꥓ꥁꥋꤰ꥓ ꤴꥎ ꤼ꥓ꤽꥊ.
```
and how it should appear
<img width="333" height="53" alt="Image" src="https://github.com/user-attachments/assets/ebf49a63-c03c-49a3-a0e7-7dff87143bfa" />
